codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -518,7 +518,7 @@ public void onPushPromiseRead(ChannelHandlerContext ctx, int streamId, int promi
         public void onGoAwayRead(ChannelHandlerContext ctx, int lastStreamId, long errorCode, ByteBuf debugData)
                 throws Http2Exception {
             // Don't allow any more connections to be created.
-            connection.goAwayReceived(lastStreamId);
+            connection.goAwayReceived(lastStreamId, errorCode, debugData);
```

Comment:
I wonder if this should come after we call `listener.onGoAwayRead`.  Anyone have thoughts on this? Do we care?